Prox/Prox/Data/PlacesController.swift

```diff
@@ -48,28 +48,38 @@ class PlacesProvider {
         var placeWithEvent: Place?
         var eventForPlace: Event?
         let lock = NSLock()
+        var placeReturned = false
+        var eventReturned = false
         place(forKey: placeKey) { place in
+            placeReturned = true
             defer {
                 lock.unlock()
             }
             lock.lock()
             guard let foundPlace = place,
                 let event = eventForPlace else {
                     placeWithEvent = place
+                    if (eventReturned) {
+                        callback(nil)
+                    }
                     return
             }
             foundPlace.events.append(event)
             callback(foundPlace)
         }
 
         eventProvider.event(forKey: eventKey) { event in
+            eventReturned = true
```

Comment:
Nit: Might be safer to move this after the lock.lock() call since this variable is being written two from potentially two different threads at different times.